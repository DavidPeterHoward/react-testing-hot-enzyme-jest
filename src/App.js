import React from 'react';
import styled, { css } from 'styled-components/macro';
import { hot } from 'react-hot-loader';

import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a>
          Learn React
        </a>
      </header>
    </div>
  );
}

export default hot(module)(App);