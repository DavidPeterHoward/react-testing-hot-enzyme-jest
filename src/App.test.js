
import React from 'react';
import { shallow, mount } from 'enzyme';
import puppeteer from "puppeteer";
import { toMatchImageSnapshot } from "jest-image-snapshot";
expect.extend({ toMatchImageSnapshot });

import App from './App';

// Smoke Test
it('renders without crashing', () => {
  shallow(<App />);
});

// Component Test
it("renders learn react link", () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find("a").text()).toEqual("Learn React");
});

// Snapshot Test
it("works", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

// UI Image Regression Tests
describe("jest-image-snapshot usage with an image received from puppeteer", () => {
  let browser;

  beforeAll(async () => {
    browser = await puppeteer.launch();
  });

  it("works", async () => {
    const page = await browser.newPage();
    await page.goto("http://localhost:3000");
    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot();
  });

  afterAll(async () => {
    await browser.close();
  });
});
