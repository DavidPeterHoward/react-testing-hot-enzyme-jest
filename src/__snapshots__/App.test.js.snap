// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a>
      Learn React
    </a>
  </header>
</div>
`;
